bioneuron_oracle/tests/test_decoders.py

```diff
@@ -117,79 +102,71 @@ def test_new_LIF_old_decoders(plt):
-                            seed=bio_seed, neuron_type=BahlNeuron())
-        # lif = nengo.Ensemble(n_neurons=bio_neurons, dimensions=dim, 
-        #                     neuron_type=nengo.LIF(), seed=bio_seed)
-        direct = nengo.Ensemble(n_neurons=1, dimensions=dim, 
+                             seed=pre_seed, neuron_type=nengo.LIF())
+        bio = nengo.Ensemble(n_neurons=bio_neurons, dimensions=dim,
+                             seed=bio_seed, neuron_type=BahlNeuron())
+        direct = nengo.Ensemble(n_neurons=1, dimensions=dim,
                                 neuron_type=nengo.Direct(),)
 
         nengo.Connection(stim, pre, synapse=None)
         nengo.Connection(pre, bio, synapse=tau_neuron, weights_bias_conn=True)
-        # nengo.Connection(pre,lif,synapse=tau_nengo)
         nengo.Connection(stim, direct, synapse=tau_nengo)
 
-        probe_stim = nengo.Probe(stim, synapse=None)
-        # probe_pre = nengo.Probe(pre,synapse=tau_nengo)
-        # probe_lif = nengo.Probe(lif,synapse=tau_nengo)
         probe_direct = nengo.Probe(direct, synapse=tau_nengo)
-        # probe_pre_spikes = nengo.Probe(pre.neurons,'spikes')
         probe_bio_spikes = nengo.Probe(bio.neurons, 'spikes')
-        # probe_lif_spikes = nengo.Probe(lif.neurons,'spikes')
-        
-    with nengo.Simulator(model,dt=dt_nengo) as sim:
+
+    with nengo.Simulator(model, dt=dt_nengo) as sim:
         sim.run(t_final)
 
-    # todo: call NEURON garbage collection
+    # TODO: call NEURON garbage collection
 
     # Generate a new decoding using the old decoders and new activities
     lpf = nengo.Lowpass(tau_nengo)
     solver = nengo.solvers.LstsqL2(reg=0.01)
     act_bio = lpf.filt(sim.data[probe_bio_spikes], dt=dt_nengo)
-    xhat_bio_mixed=np.dot(act_bio,decoders_bio_old)
-    rmse_bio_mixed=rmse_bio_old=rmse(sim.data[probe_direct],xhat_bio_mixed)
+    xhat_bio_mixed = np.dot(act_bio, decoders_bio_old)
+    rmse_bio_mixed = rmse(sim.data[probe_direct], xhat_bio_mixed)
     # Generate a new decoding using the new decoders and new activities
     decoders_bio_new, info = solver(act_bio, sim.data[probe_direct])
-    xhat_bio_new=np.dot(act_bio,decoders_bio_new)
-    rmse_bio_new=rmse_bio_old=rmse(sim.data[probe_direct],xhat_bio_new)
+    xhat_bio_new = np.dot(act_bio, decoders_bio_new)
+    rmse_bio_new = rmse(sim.data[probe_direct], xhat_bio_new)
     # Calculate the RMSE of the mixed and new decoding
-    rmse_compare=rmse_bio_old=rmse(xhat_bio_mixed,xhat_bio_new)
-
-    plt.subplot(1,1,1)
-    plt.plot(sim.trange(),xhat_bio_old,label='old, rmse=%.5f'%rmse_bio_old)
-    plt.plot(sim.trange(),xhat_bio_mixed,label='mixed, rmse=%.5f'%rmse_bio_mixed)
-    plt.plot(sim.trange(),xhat_bio_new,label='new, rmse=%.5f'%rmse_bio_new)
-    # plt.plot(sim.trange(),sim.data[probe_lif],label='lif, rmse=%.5f'%rmse_lif)
-    plt.plot(sim.trange(),sim.data[probe_direct],label='direct')
+    rmse_bio_compare = rmse(xhat_bio_mixed, xhat_bio_new)
```

Comment:
FYI you were repeatedly overwriting `rmse_bio_old`, and so the numbers in your plots for `old` would have been wrong. This was done a few times in this file.